internal/sync.go

```diff
@@ -296,3 +307,34 @@ func (s *syncGSuite) ignoreGroup(name string) bool {
 
 	return false
 }
+
+func (s *syncGSuite) allowGroup(name string) bool {
+	if len(s.cfg.AllowGroups) == 0 {
+		return true
+	}
+
+	for _, g := range s.cfg.AllowGroups {
+		if strings.HasPrefix(name, g) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (s *syncGSuite) allowPattern(name string) bool {
+	if len(s.cfg.AllowGroups) == 0 {
+		return true
+	}
+	for _, p := range s.cfg.AllowPattern {
+		if p == "" {
+			return true
+		}
+
+		re := regexp.MustCompile(p)
```

Comment:
Is this going to be a performance problem with lots of patterns compiled and applied at each user? Should we really be preprocessing the patterns once to compile them (newish to go)